messaging/src/main/java/org/axonframework/commandhandling/gateway/ExponentialBackOffIntervalRetryScheduler.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2010-2019. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandMessage;
+
+import java.util.List;
+
+import static org.axonframework.common.BuilderUtils.assertStrictPositive;
+
+/**
+ * A RetryScheduler that uses a backoff strategy, retrying commands at increasing intervals when they fail because of
+ * an exception that is not explicitly non-transient. Checked exceptions are considered non-transient and will not
+ * result in a retry.
+ *
+ * @author Bert Laverman
+ * @since 4.2
+ */
+public class ExponentialBackOffIntervalRetryScheduler extends AbstractRetryScheduler {
+
+    private static final long DEFAULT_BACKOFF_FACTOR = 100L;
+
+    private final long backoffFactor;
+
+    /**
+     * Instantiate an {@link ExponentialBackOffIntervalRetryScheduler}. the settings are copied from the
+     * {@link Builder} and validated.
+     *
+     * @param builder
+     */
+    protected ExponentialBackOffIntervalRetryScheduler(Builder builder) {
+        super(builder);
+        builder.validate();
+
+        this.backoffFactor = builder.backoffFactor;
+    }
+
+    /**
+     * Instantiate a Builder to be able to create a {@link ExponentialBackOffIntervalRetryScheduler}.
+     * <p>
+     * The default for {@link Builder#backoffFactor} is set to 100ms, and must be at least 1.
+     *
+     * @return a Builder to be able to create a {@link ExponentialBackOffIntervalRetryScheduler}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    protected long computeRetryInterval(CommandMessage commandMessage,
+                                        RuntimeException lastFailure,
+                                        List<Class<? extends Throwable>[]> failures) {
+        final int retryCount = failures.size();
+        return backoffFactor * (1L << (retryCount - 1));
```

Comment:
how about Math.pow(retryCount - 1, configurableRate)?

Reply:
This implementation does what the class name promises. Using e.g. a function instead is admittedly more flexible, but not what we were chasing here. Open a new issue to start that discussion please.

Reply:
Sure thanks.. Nice addition.

Reply:
@gjmwoods "Math.pow(retryCount - 1, configurableRate)" == "(retryCount-1)^configurableRate". That is something completely different from "configurableRate * 2^(retryCount-1)". If your intent is to ask me why I didn't use `Math::pow` then the answer is that I am computing a power of two, and going through a conversion to double and back, combined with a function for raising any base to any power, is a waste of time and precision.

Reply:
@bert-laverman I think gjmwoods's proposal is something like this,
`backoffFactor * Math.pow(configurableRate, retryCount - 1);`
This way it will not be always power of two, it will be power of 'configurableRate'.